app/src/store/data.ts

```diff
@@ -37,94 +48,288 @@ const lastBlockTimestamp = ref<number>(0);
 
 const grants = ref<Grant[]>();
 const grantMetadata = ref<Record<string, GrantMetadataResolution>>({});
-const distributions = ref<GrantsDistribution>();
+const grantContributions = ref<{ [hash: string]: Contribution }>();
 
-const grantRounds = ref<GrantRounds>();
+const grantRounds = ref<GrantRound[]>();
 const grantRoundMetadata = ref<Record<string, GrantRoundMetadataResolution>>({});
+const grantRoundsCLRData = ref<{ [grantRound: string]: GrantRoundCLR }>();
 
 // --- Store methods and exports ---
 export default function useDataStore() {
+  // LocalStorage keys/prefixes
+  const allGrantsKey = 'AllGrants';
+  const allGrantRoundsKey = 'AllGrantRounds';
+  const grantRoundKeyPrefix = 'GrantRound-';
+  const contributionsKey = 'Contributions';
+  const grantRoundsCLRDataKeyPrefix = 'GrantRoundsGrantData-';
+  const trustBonusKey = 'TrustBonus';
+
   /**
    * @notice Called each block to poll for data, but can also be called on-demand, e.g. after user submits a transaction
    */
-  async function poll() {
+  async function rawPoll(forceRefresh = false) {
     if (!multicall.value || !registry.value || !roundManager.value) return;
 
     // Define calls to be read using multicall
     const calls = [
       { target: MULTICALL_ADDRESS, callData: multicall.value.interface.encodeFunctionData('getCurrentBlockTimestamp') },
-      { target: GRANT_REGISTRY_ADDRESS, callData: registry.value.interface.encodeFunctionData('getAllGrants') },
```

Comment:
Do we even need this first round of multicalls here? I don't think we use this timestamp for anything and can probably get rid of it. Seems like an artifact of the default dummy `data.ts` code we had when we initialized the project

Reply:
This has been removed 🙌